java/flight/flight-sql/src/test/java/org/apache/arrow/flight/TestFlightSql.java

```diff
@@ -0,0 +1,260 @@
+          }
+        } else if (fieldVector instanceof IntVector) {
+          for (int rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+            actualIntResults.add(((IntVector) fieldVector).get(rowIndex));
+          }
+        }
+      }
+    }
+    stream.getRoot().clear();
+
+    assertEquals(Arrays.asList("one", "zero", "negative one"), actualStringResults);
+    assertEquals(Arrays.asList(1, 0, -1), actualIntResults);
+
+    final Iterator<Result> closePreparedStatementResults = client.doAction(new Action("ClosePreparedStatement",
+            Any.pack(ActionClosePreparedStatementRequest
+                    .newBuilder()
+                    .setPreparedStatementHandleBytes(preparedStatementResult.getPreparedStatementHandle())
+                    .build())
+                    .toByteArray()));
+    assertFalse(closePreparedStatementResults.hasNext());
+  }
+
+  @Test
+  public void testSimplePrepStmtWithFlightSqlClientUtils() throws Exception {
+    final FlightSqlClientUtils.FlightSqlPreparedStatement preparedStatement =
+            getPreparedStatement(client, "Select * from intTable");
+
+    final Schema actualSchema = preparedStatement.getResultSetSchema();
+    assertEquals(SCHEMA_INT_TABLE, actualSchema);
+
+    final FlightInfo info = preparedStatement.executeQuery();
+    assertEquals(SCHEMA_INT_TABLE, info.getSchema());
+
+    final FlightStream stream = client.getStream(info.getEndpoints().get(0).getTicket());
+    assertEquals(SCHEMA_INT_TABLE, stream.getSchema());
+
+    List<String> actualStringResults = new ArrayList<>();
+    List<Integer> actualIntResults = new ArrayList<>();
+    while (stream.next()) {
+      final VectorSchemaRoot root = stream.getRoot();
+      final long rowCount = root.getRowCount();
+
+      for (Field field : root.getSchema().getFields()) {
+        final FieldVector fieldVector = root.getVector(field.getName());
+
+        if (fieldVector instanceof VarCharVector) {
+
+          final ElementAddressableVectorIterator<VarCharVector> it =
+                  new ElementAddressableVectorIterator<>((VarCharVector) fieldVector);
+
+          for (int rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+            final ArrowBufPointer pt = it.next();
+            final byte[] bytes = new byte[(int) pt.getLength()];
+            pt.getBuf().getBytes(pt.getOffset(), bytes);
+
+            actualStringResults.add(new String(bytes, StandardCharsets.UTF_8));
+          }
+        } else if (fieldVector instanceof IntVector) {
+          for (int rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+            actualIntResults.add(((IntVector) fieldVector).get(rowIndex));
+          }
+        }
+      }
+    }
+    stream.getRoot().clear();
+
+    assertEquals(Arrays.asList("one", "zero", "negative one"), actualStringResults);
+    assertEquals(Arrays.asList(1, 0, -1), actualIntResults);
+
+    AutoCloseables.close(preparedStatement);
```

Comment:
try-with-resources instead of explicit close.

Reply:
That was done intentionally to allow for testing.